README.md

```diff
@@ -218,6 +218,12 @@ Note that First-Party Sets also gives browsers the opportunity to group per-site
 those at `chrome://settings/content/all`) by the “first-party” boundary instead of eTLD+1, which is 
 not always the correct site boundary.
 
+# Domain Schemes
+
+User agents must "normalize" WebSocket schemes to HTTP(S) when determining whether a particular domain is a member of a First-Party Set. I.e. `ws://` must be mapped to `http://`, and `wss://` must be mapped to `https://`, before the lookup is performed.
```

Comment:
This is already built into Fetch across all WebSockets-related logic as https://fetch.spec.whatwg.org/#websocket-opening-handshake. If this is just intended as a clarification, probably should be phrased as such and cite the spec.

Reply:
This might be a miscommunication - I'm not talking about when the browser is opening the connection, I'm talking about when the user-agent does a lookup into its collection of First-Party Sets (see https://crrev.com/c/3182786 for example). My impression was that the existing text in the Fetch spec wouldn't necessarily apply to that algorithm.

Does that make sense?

Reply:
What lookup points are you envisioning? I would expect that all to be inside Fetch. (WebSocket requests don't commit documents or anything, so there are no wss document origins.) If it's inside Fetch, that mapping step means that FPS will never see wss URLs, only https URLs.

https://crrev.com/c/3182786 is one such example. It looks like that was about cookies, and cookies are handled [inside Fetch](https://fetch.spec.whatwg.org/#http-network-or-cache-fetch). So, by spec, the cookie logic already sees the mapped URL, not the wss one.

The reason we needed to map separately in Chromium is an implementation quirk: Chromium happens not to map at that point, which means it needs to separately map it elsewhere, to match the specification. But that is specific to Chromium and not a spec issue. (This is fine. The spec describes the system's behavior, not a precise implementation. Implementations may be structured differently, as long as they give the same behavior. Though it may be worth aligning Chromium to the spec's model, just so this sort of thing is simpler.)

Reply:
Hm, I see. So the argument is that any network request that includes cookies must have gone through the Fetch algorithm (or something that implements the Fetch spec, at any rate), and therefore must already treat `wss://` URLs as `https://` URLs for the purpose of cookies.

(It was a little bit confusing that Chromium's implementation is organized very differently than the spec, but I assume that's due to historical reasons.)

In that case this would just be a clarification (if it's present at all); I'll try rewording it so that's clear.